torch/nn/utils/parametrize.py

```diff
@@ -263,6 +269,8 @@ def right_inverse(self, X: Tensor) -> Tensor
     if is_parametrized(module, tensor_name):
         # Just add the new parametrization to the parametrization list
         module.parametrizations[tensor_name].append(parametrization)  # type: ignore[index, union-attr]
+        # If allow_resize was True in previous parametrization, keep it enabled
+        module.parametrizations[tensor_name].allow_resize |= allow_resize  # type: ignore[index, union-attr]
```

Comment:
```suggestion
        module.parametrizations[tensor_name].allow_resize = allow_resize  # type: ignore[index, union-attr]
```
as the `allow_resize` parameter just affects the output of the last registered parametrisation. This way the mechanism is always opt-in / opt-out.

Reply:
The problem with overriding the `allow_resize` is that if you register multiple, when only the early ones are resizing, it will break it. Consider an example

```
register_parametrization(model.linear, 'weight', ResizingParametrization, allow_resize=True)
register_parametrization(model.linear, 'weight', NonResizingParametrization)
```

In the example above, the `allow_resize` is set to `False`, but because of the sequential nature, the first parametrization will resize the tensor, and the whole thing will throw an error.

I think the behavior should be that once you allow resizing, the whole parametrization list becomes unsafe.

Reply:
Ah, I see what you mean, yes.

It also makes more sense to leave it the way it is now when if we make this into an `unsafe` check. Once you do not check the correctness of one of the parametrisations, you break the invariants and these may not hold from there on.

This point should be documented though.

Reply:
Hi @Lezcano, what do you mean by making this into an `unsafe` check/renaming this flag to `unsafe`? Do you mean like renaming `allow_resize` to `unsafe` and updating the documentation to be more generic e.g. "if enabled, parametrization invariants may not hold anymore"?

Reply:
Yes, rename `allow_resize` to `unsafe` and turn off quite a few other checks. In that PR mentioned above, we perform quite a few checks both in `ParametrizationList.__init__` and in `register_parametrization`.

In particular, in the file
https://github.com/pytorch/pytorch/blob/f675a3c020d638b64c988457e715c2e561667cb8/torch/nn/utils/parametrize.py
Lines 455-493, should be move them under an `if not unsafe:`.
Same with L169-188 (and also remove the `not self.allow_resize` in L183).

We then should modify the messages raised by these exceptions to announce to the user that this `unsafe` flag exists.